palace/models/timeoperator.cpp

```diff
@@ -130,12 +152,22 @@ class TimeDependentFirstOrderOperator : public mfem::TimeDependentOperator
       du = 0.0;
     }
     FormRHS(u, RHS);
-    Vector du1(du.GetData() + 0, size_E);
-    Vector du2(du.GetData() + size_E, size_E);
-    Vector rhs1(RHS.GetData() + 0, size_E);
-    Vector rhs2(RHS.GetData() + size_E, size_E);
+
+    Vector du1, du2, rhs1, rhs2;
+    du1.UseDevice(true);
+    du2.UseDevice(true);
+    rhs1.UseDevice(true);
+    rhs2.UseDevice(true);
+    du.GetSubVector(idx1, du1);
+    du.GetSubVector(idx2, du2);
+    RHS.GetSubVector(idx1, rhs1);
+    RHS.GetSubVector(idx2, rhs2);
```

Comment:
Why do you need `idx1` or `idx2` here rather than the approach taken in [`ComplexVector`](https://github.com/awslabs/palace/blob/main/palace/linalg/vector.cpp#L54-L61) of using `ReadWrite()`?

```
du.ReadWrite();
Vector du1(du.GetData() + 0, size_E), du2(du.GetData() + size_E, size_E);
RHS.ReadWrite();
Vector rhs1(du.GetData() + 0, size_E), rhs2(du.GetData() + size_E, size_E);
```
also probably worth making `RHS` into `rhs` or `rhsX` into `RHSX` for consistency.

Reply:
I tried that first but I was getting errors like this when running on GPU:

> Verification failed: (it != maps->memories.end()) is false:
 --> host pointer is not registered: h_ptr = 0x2af94870
 ... in function: static void mfem::MemoryManager::CheckHostMemoryType_(mfem::MemoryType, void*, bool)
 ... in file: /data/home/simlap/palace/build/extern/mfem/general/mem_manager.cpp:1714

Then I saw this page https://mfem.org/gpu-support/ warning against `GetData()` on GPU.